lib/scnnr/client.rb

```diff
@@ -20,66 +20,121 @@ def config
     end
 
     def recognize_image(image, options = {})
-      options = merge_options options
-      PollingManager.start(self, options) do |opts|
-        uri = construct_uri('recognitions', %i[timeout public], opts)
-        response = post_connection(uri, opts).send_stream(image)
-        Response.new(response).build_recognition
+      recognize(options) do |opts|
+        recognize_image_with_file(image, opts)
       end
     end
 
     def recognize_url(url, options = {})
-      options = merge_options options
-      PollingManager.start(self, options) do |opts|
-        uri = construct_uri('remote/recognitions', %i[timeout force], opts)
-        response = post_connection(uri, opts).send_json({ url: url })
-        Response.new(response).build_recognition
+      recognize(options) do |opts|
+        recognize_image_with_url(url, opts)
       end
     end
 
     def fetch(recognition_id, options = {})
-      options = merge_options options
-      return request(recognition_id, options) if options.delete(:polling) == false
+      options = default_options.merge options
+      return get_recognition(recognition_id, options) if without_timeout?(options[:timeout])
 
-      PollingManager.new(options.delete(:timeout)).polling(self, recognition_id, options)
+      validate_timeout(options[:timeout])
+      timeout_at = Scnnr::PollingUtils.timeout_at(options[:timeout])
+
+      recognize_poll(recognition_id, timeout_at, options)
     end
 
     def coordinate(category, labels, taste = {}, options = {})
-      options = merge_options options
+      options = default_options.merge options
       uri = construct_uri('coordinates', %i[target], options)
       payload = {
         item: { category: category, labels: labels },
         taste: TASTES.each_with_object({}) { |key, memo| memo[key] = taste[key] if taste[key] },
       }
-      response = post_connection(uri, options).send_json(payload)
+      response = post(uri, payload, options)
       Response.new(response).build_coordinate
     end
 
     private
 
-    def merge_options(options = {})
-      self.config.to_h.merge(options)
+    def recognize(options, &block)
```

Comment:
How about renaming this method? Because `#recognize` doesn't really recognize anything and actually `block` does it.
By this, we need to call this like `recognize(options) { |opts| recognize_image_with_url(url, opts) }`. To understand how a URL is recognized, an engineer needs to read its implementation, but it's hard to guess whether `#recognize` does it or `#recognize_image_with_url` does.

Reply:
sounds good, thanks